### DISCUSSION TITLE
Rules from remote location

### DISCUSSION THREAD

**Discussion by a7s1h1** [detekt/detekt#3857] [Q&A] [answered]
At the moment I keep my custom detekt-config.yml in project resources, referenced in build.gradle.kts:
`detekt {    config = files("$rootDir/src/main/resources/detekt-config.yml") }`
But in order to share the same rules between different projects, it would be convenient to have the rules in some remote location to take them from there from any place, something like
`detekt {    config = files("https://remote-location/detekt-config.yml") }`
How can this be achieved?

**Comment by BraisGabin** [comment, marked as answer]
You can create your own gradle task to download a file from the network. Then you just need to add your task as a prerequisit to run the detekt task and done.

**Comment by a7s1h1** [comment]
> You can create your own gradle task to download a file from the network. Then you just need to add your task as a prerequisit to run the detekt task and done.

Thanks for the advice. It works, thought I had either write ugly download task myself, or use de.undercouch.download plugin.
I have chosen the second, but are there any better approaches to download files via task?

**Comment by cortinico** [comment]
> How can this be achieved?

You can also mount a remote location locally and keep on using `files()`. E.g. you could mount a remote SSH location with SSHFS or similar solutions
